src/features/Folding.ts

```diff
@@ -0,0 +1,588 @@
+import * as path from "path";
```

Comment:
What's the story on source file copyright notices?  Every other .ts file seems to have this copyright notice on the first line:
```
/*---------------------------------------------------------
 * Copyright (C) Microsoft Corporation. All rights reserved.
 *--------------------------------------------------------*/

```
Even those contributed by community members.

Reply:
❓ Contributing guide/s makes no mention.  I've already signed the CLA.

Reply:
I, as a non-Microsoft employee, shouldn't add that, as I don't represent them.

Reply:
I don't mention this as a blocker but an inquiry as to what the policy is.  Back in January I went through this code-base and cleaned up source to pass tslint as part of effort to achieve some form of engineering compliance  (#1086).  I'm curious if that "compliance" has anything to say about copyright notices.

Reply:
> I, as a non-Microsoft employee, shouldn't add that, as I don't represent them.

FWIW the copyright is their's either way, and ultimately the one representing Microsoft is whomever publishes the release.

I view it more as "informing of a copyright" instead of "declaring copyright".

That said, I'm not sure what the point of it is - at least in an MIT licensed project.  Maybe @tylerl0706 or @rjmholt could check if there is a policy on this?  If there isn't, my vote would be to remove it or at least omit it going forward.

(This is just a general note, definitely shouldn't hold up the PR)

Reply:
The copyright header is definitely required. If you don't want to add it, I can do a quick amend.

Reply:
This is one of those "I don't make the rules, I just (have to) follow them

Reply:
We should definitely add a note in the contribution guide...

Reply:
added the headers

Reply:
Btw, I looked at our contriibution guide and was about to add a note but wasn't sure of where to add it or how to phrase it. I also notice that PowerShell doesn't seem to have a note about it either, only in their PR checklist.

Reply:
We could add it to the PR checklist :)

Reply:
Offtopic - @SeeminglyScience - Ceeding copyright to Microsoft is fine.  It was more the "All rights reserved".  IANAL, but I didn't feel I was in position to represent to Microsoft to say what rights other people had. Again this all moot (Heh, I used legal term!) due to me signing the CLA.

Reply:
@tylerl0706 @rjmholt Raised #1394 to add automation to enforce header rule.